### PR TITLE
Initial drop of hand-written code for Vision V1

### PR DESCRIPTION
**Pull request by jskeet**
This is reasonably complete, but the error handling may well need to
change. If we can reasonably expect that there will never be an
error *and* annotations in a response for a single image, we can
throw an appropriate exception in all the "process single image"
methods.

### CODE REVIEW

File: apis/Google.Cloud.Vision.V1/Google.Cloud.Vision.V1.Tests/FakeImageAnnotator.cs [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    /// with all relevant fields filled in; it then examines the request to find out which aspects
+    /// were requested (faces, text etc).
+    /// </summary>
+    public class FakeImageAnnotator : ImageAnnotatorClient
+    {
+        public override BatchAnnotateImagesResponse BatchAnnotateImages(IEnumerable<AnnotateImageRequest> requests, CallSettings callSettings = null)
+            => new BatchAnnotateImagesResponse { Responses = { requests.Select(r => ProcessRequest(r)) } };
+
+        public override Task<BatchAnnotateImagesResponse> BatchAnnotateImagesAsync(IEnumerable<AnnotateImageRequest> requests, CallSettings callSettings = null)
+        {
```
Comment (chrisdunelm) [inline apis/Google.Cloud.Vision.V1/Google.Cloud.Vision.V1.Tests/FakeImageAnnotator.cs]: expression body?

Comment (jskeet) [inline apis/Google.Cloud.Vision.V1/Google.Cloud.Vision.V1.Tests/FakeImageAnnotator.cs]: Done.

File: apis/Google.Cloud.Vision.V1/Google.Cloud.Vision.V1/AnnotationResult.cs [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    /// </summary>
+    /// <typeparam name="T">The type of annotation returned.</typeparam>
+    public sealed class AnnotationResult<T> : IReadOnlyList<T>
+    {
+        private readonly RepeatedField<T> annotations;
+
+        /// <summary>
+        /// The error status in the response, or <c>null</c> if the call was successful.
+        /// </summary>
+        public Rpc.Status Error { get; }
```
Comment (chrisdunelm) [inline apis/Google.Cloud.Vision.V1/Google.Cloud.Vision.V1/AnnotationResult.cs]: This order doesn't seem quite right, should this property be after the constructor?

Comment (jskeet) [inline apis/Google.Cloud.Vision.V1/Google.Cloud.Vision.V1/AnnotationResult.cs]: We usually declare properties before the constructor, I believe. (I generally do, anyway.) Admittedly `Count` doesn't follow that, but that's delegating...

One reason for that is that fields go before the constructor - and auto-implemented properties are declaring a field as well as the property. Basically I like to see all the actual state of a class at the top. But YMMV :)

File: apis/Google.Pubsub.V1/Google.Pubsub.V1.Snippets/PublisherClientSnippets.cs [hunk 3]
```diff
@@ -198,5 +200,19 @@ public void DeleteTopic()
             Console.WriteLine($"Deleted {topicName}");
             // End snippet
         }
+
+        [Fact]
+        public void Foo()
```
Comment (chrisdunelm) [inline apis/Google.Pubsub.V1/Google.Pubsub.V1.Snippets/PublisherClientSnippets.cs]: Should this be here?

File: apis/Google.Cloud.Vision.V1/Google.Cloud.Vision.V1/AnnotationResult.cs (hunk 2 above)
Comment (chrisdunelm) [inline apis/Google.Cloud.Vision.V1/Google.Cloud.Vision.V1/AnnotationResult.cs]: OK, although I don't really like the property split before/after the constructor...

**Comment by chrisdunelm** [comment]
LGTM (apart from the Foo test)

File: apis/Google.Pubsub.V1/Google.Pubsub.V1.Snippets/PublisherClientSnippets.cs (hunk 3 above)
Comment (jskeet) [inline apis/Google.Pubsub.V1/Google.Pubsub.V1.Snippets/PublisherClientSnippets.cs]: Whoops, no :)

**Pull request merged**